### PR TITLE
Notification Dropdown : Migrate links to Symfony routes

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Notification Dropdown : Migrate links to Symfony routes. Thanks to @matks for the base in #19385.
| Type?             | improvement
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | Relative to #16592
| How to test?      | * Build assets<br>* From the Front Office go to Order history -> Details and then select product, enter message and send it.<br>* Login to the Back Office (Legacy Page) -> Notification dropdown toggle (bell icon top right corner) -> all links target migrated pages<br>* Go to database in the `employee` tab : `id_last_order`, `id_last_customer_message` and `id_last_customer` and set values to 0 (That reset the notificaition)<br>* Got to page Order (Migrated Page) -> Notification dropdown toggle (bell icon top right corner) -> all links target migrated pages

**:notebook: BC Breaks:**
* Added two parameters `string $customerThreadViewUrl` and `string $orderViewUrl`to the constructor `\PrestaShop\PrestaShop\Core\Domain\Notification\QueryResult\NotificationResult`

### CODE REVIEW

**Review by NeOMakinG — DISMISSED** [review]

File: js/admin/notifications.js [hunk 1]
```diff
@@ -79,11 +78,10 @@ function renderCustomerNotification(value) {
 }
 
 function renderMessageNotification(value) {
-  const query = `tab=AdminCustomerThreads&token=${token_admin_customer_threads}&viewcustomer_thread&id_customer_thread=${value.id_customer_thread}`;
   const company = value.company !== '' ? ` (${value.company})` : '';
   return `
-    <a class="notif" href="${baseAdminDir}index.php?${query}">
-      <span class="message-notification-status ${value.status}">
+    <a class="notif" href="${value.customer_thread_view_url}">
+      <span class="message-notification-status customer_thread_view_url{value.status}">
```
Comment (atomiix) [inline js/admin/notifications.js]: I'm not sure to understand here 🤔
```suggestion
      <span class="message-notification-status ${value.status}">
```

Comment (Progi1984) [reply js/admin/notifications.js]: Fixed @atomiix

**Review by NeOMakinG — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

**Comment by atomiix** [comment]
@Progi1984 it seems that there's a conflicted file :/

**Comment by Progi1984** [comment]
@atomiix & @NeOMakinG It's rebased.

**Review by NeOMakinG — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @Progi1984 ,

Tested on develop, PHP 7.4.

Tested the following:
- message sent from FO > Customer account > Order history
- message sent from FO > contact form (as guest)
- message sent from FO > contact form (as customer with/without order)

From BO (migrated page and legacy), all links are redirected to migrated page (shop.com/adminXXX/index.php/sell/customer-service/customer-threads/)

It is QA ✅ 
Thanks!

**Pull request merged**